crates/stages/src/db.rs

```diff
@@ -164,6 +165,21 @@ where
         Ok((prev_body.start_tx_id + prev_body.tx_count, last_transition))
     }
 
+    /// Get the result vector with range `[start_key:end_key)`.
+    /// Please note that `start_key` is inclusive and `end_key` is exclusive.
+    pub(crate) fn get_range<T: Table>(&self, start_key: T::Key, end_key: T::Key) -> VecPairResult<T>
+    where
+        T: Table,
+    {
+        self.cursor_read::<T>()?
+            .walk(start_key)?
+            .take_while(|res| match res {
+                Ok((key, _)) => *key < end_key,
+                Err(_) => false,
+            })
+            .collect::<Result<Vec<_>, _>>()
```

Comment:
perf: this would load everything into mem as opposed to walking and reading one by one, right @rakita? maybe we should return `Vec<PairResult<_>>`

Reply:
Yes, to my understanding, it loads everything into mem. As far as I know, if we do `collect()`, it might load everything into mem even though we return `Vec<PairResult<_>>`.

For perf, I tried to return `Walker<>` but failed because it's part of `DbCursor<>`.

### Positive thinking for my changes even though considering perf,
* All use cases that already do `collect()` except `sender_recovery.rs` that does `chunks()` iterator.
* We could remove this change from `sender_recovery.rs`.

### Negative thinking for my changes considering perf,
* I think fundamental solution is that revises `DbCursor<>` to add an interface
```
    fn walk_while<'cursor>(
        &'cursor mut self,
        start_key: T::Key,
        end_key: T::Key,
    ) -> Result<Walker<'cursor, 'tx, T, Self>, Error>
```    
* This approach is needed to change `DbCursor<>` and it could broadly affect on reth. 
* Just as a newbie contributor, I'm little bit affraid to approach this method.

I need feedback and opinion. Thanks,

Reply:
And also please note my comment on the function name, https://github.com/paradigmxyz/reth/pull/942#discussion_r1082614212

Reply:
@rkrasiuk returning `Vec<PairResult<_>>` would still result in this loading everything into mem.

I think the solution is to move this to the cursor as `walk_range` instead. It would require us to do `tx.cursor_read::<Table>().walk_range(..)` as opposed to `tx.get_range` but imo this is ok given perf concern.

Reply:
I am using it like that, I know how much I want to load and when reading the table I read it in batches to store it in memory.

Reply:
How come? You could just use the iterator directly in the storage hashing stage which would be way easier on memory compared to loading everything into a `Vec` that is immediately turned into an iterator again

Reply:
For reference I am talking about these lines: https://github.com/paradigmxyz/reth/blob/a1ad688c52fe6b61b07be36628589fbd96c6c1db/crates/stages/src/stages/hashing_storage.rs#L94-L114

The collect on L99 can be avoided, the one on L114 can't (but the memory footprint there is potentially *a lot* smaller)

Reply:
Similarly the `collect` can be avoided here: https://github.com/paradigmxyz/reth/blob/a1ad688c52fe6b61b07be36628589fbd96c6c1db/crates/stages/src/stages/hashing_storage.rs#L118-L130

Reply:
It is faster to load in bundles and have an iterator load as much as possible.